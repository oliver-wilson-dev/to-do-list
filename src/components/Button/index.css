@import "../../styles/variables.css";

.button {
    border: none;
    text-align: right;
    font-size: 18px;
    background-color: var(--primaryBackgroundColour);
    border-radius: inherit;
}
