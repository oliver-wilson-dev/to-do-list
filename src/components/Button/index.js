import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

const Button = ({
  onClick, children, additionalStyles
}) => (
  <button className={`${styles.button} ${additionalStyles || ''}`} onClick={onClick} type="button">
    {children}
  </button>
);

Button.propTypes = {
  additionalStyles: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

export default Button;
