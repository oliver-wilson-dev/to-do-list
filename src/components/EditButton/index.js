import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

const EditButton = ({
  setEditState, id, editing, description
}) => {
  const onClick = () => setEditState({ id, editState: !editing, description });

  return (
    <Button onClick={onClick}>
      {editing
        ? <span role="img" aria-label="apply changes">✅</span>
        : <span role="img" aria-label="edit this task">✏️</span>
      }
    </Button>
  );
};

EditButton.propTypes = {
  setEditState: PropTypes.func.isRequired,
  id: PropTypes.symbol.isRequired,
  editing: PropTypes.bool.isRequired,
  description: PropTypes.string.isRequired
};

export default EditButton;
