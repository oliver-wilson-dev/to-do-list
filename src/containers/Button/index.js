import { connect } from 'react-redux';
import Button from '../../components/Button';
import incrementCount from '../../state/actions/incrementCount';

const mapDispatchToProps = dispatch => ({
  incrementCount: () => dispatch(incrementCount())
});

export default connect(undefined, mapDispatchToProps)(Button);
