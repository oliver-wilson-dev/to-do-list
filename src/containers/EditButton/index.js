import { connect } from 'react-redux';
import EditButton from '../../components/EditButton';
import setEditState from '../../state/actions/setEditState';

const mapDispatchToProps = dispatch => ({
  setEditState:
  ({ id, editState, description }) => dispatch(setEditState({ id, editState, description }))
});

export default connect(undefined, mapDispatchToProps)(EditButton);
