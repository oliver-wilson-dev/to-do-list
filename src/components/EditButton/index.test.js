import React from 'react';
import { shallow } from 'enzyme';
import EditButton from './index';
import Button from '../Button';

jest.mock('../Button', () => {
  const Button = () => null;

  return Button;
});

const mockId = Symbol('test-id');
const mockDescription = 'test-description';
const mockSetEditState = jest.fn();
const mockEditing = false;

const defaultProps = {
  setEditState: mockSetEditState,
  id: mockId,
  editing: mockEditing,
  description: mockDescription
};

const render = overrideProps => shallow(<EditButton {...defaultProps} {...overrideProps} />);

describe('<EditButton/> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const onClickChecks = ({ mockEditing } = { mockEditing: false }) => {
    describe("when the button's onClick prop is called", () => {
      it("should call the removeTask function with the task's id", () => {
        const mockId = Symbol('test-id');
        const mockDescription = 'test-description';
        const mockSetEditState = jest.fn();

        render({
          setEditState: mockSetEditState,
          id: mockId,
          editing: mockEditing,
          description: mockDescription
        })
          .find(Button).simulate('click');

        expect(mockSetEditState)
          .toHaveBeenCalledWith({
            id: mockId,
            editState: !mockEditing,
            description: mockDescription
          });
      });
    });
  };

  describe('when the task is not being edited', () => {
    it('should render correctly', () => {
      expect(render()).toMatchSnapshot();
    });

    onClickChecks();
  });

  describe('when the task is being edited', () => {
    it('should render correctly', () => {
      expect(render({ editing: true })).toMatchSnapshot();
    });

    onClickChecks({ mockEditing: true });
  });
});
