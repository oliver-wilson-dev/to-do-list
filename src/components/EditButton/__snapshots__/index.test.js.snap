// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditButton/> component when the task is being edited should render correctly 1`] = `
<Button
  onClick={[Function]}
>
  <span
    aria-label="apply changes"
    role="img"
  >
    ✅
  </span>
</Button>
`;

exports[`<EditButton/> component when the task is not being edited should render correctly 1`] = `
<Button
  onClick={[Function]}
>
  <span
    aria-label="edit this task"
    role="img"
  >
    ✏️
  </span>
</Button>
`;
